sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala

```diff
@@ -15,14 +15,66 @@
  */
 
 package kalix.springsdk
-import akka.annotation.ApiMayChange
-import kalix.springsdk.impl.RestDeferredCall
+import kalix.javasdk.DeferredCall
+import com.google.protobuf.any.Any
 
+/**
+ * Utility to send requests to other Kalix components by composing a DeferredCall. The target component endpoint should
+ * belong to a service on the same project.
+ */
 trait KalixClient {
 
-  @ApiMayChange
-  def post[P, R](uri: String, body: P, returnType: Class[R]): RestDeferredCall[P, R]
+  /**
+   * Provides utility to do a POST HTTP request to a target endpoint belonging to a Kalix component. Such endpoint is
+   * identified by a resource path (i.e. excluding authority and scheme).
+   *
+   * Example of use:
+   * {{{
+   *     @PostMapping("/next")
+   *     public Effect<Number> nextNumber(@RequestBody Number number) {
+   *       var serviceCall = kalixClient.post("/fibonacci/next", number, Number.class);
+   *       return effects().forward(serviceCall);
+   *     }
+   * }}}
+   *
+   * @param uri
+   *   The resource path where the target endpoint will be reached at. Query parameters can be passed in as part of the
+   *   URI but should be encoded if containing special characters.
+   * @param body
+   *   The HTTP body type expected by the target endpoint
+   * @param returnType
+   *   The type returned by the target endpoint
+   * @tparam P
+   *   Type used as a body for the request
+   * @tparam R
+   *   Type returned as response from the target endpoint
+   * @return
+   *   a [[kalix.javasdk.DeferredCall]] to be used as Effect or executed
```

Comment:
Not sure if we should say that we can use a `DeferredCall` as an `Effect`. Have we said so in other parts for the documentation? 

I think we should say:
"to be used in forwards and timers or to be executed in place"